cmake/ccache.cmake

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+if(NOT MSVC)
+  find_program(CCACHE_EXECUTABLE ccache)
+  if(CCACHE_EXECUTABLE)
+    execute_process(
+      COMMAND readlink -f ${CMAKE_CXX_COMPILER}
+      OUTPUT_VARIABLE compiler_resolved_link
+      ERROR_QUIET
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(CCACHE_EXECUTABLE STREQUAL compiler_resolved_link)
+      set(WITH_CCACHE "ccache masquerades as the compiler")
+    elseif(WITH_CCACHE)
+      list(APPEND CMAKE_C_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
+      list(APPEND CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
+    endif()
+    try_append_cxx_flags("-fdebug-prefix-map=A=B" TARGET core_interface SKIP_LINK
+      IF_CHECK_PASSED "-fdebug-prefix-map=${PROJECT_SOURCE_DIR}=."
```

Comment:
Note that with 80073c75d1218759f58ceefcb41b4fbc4a3d1ecd I could right click and debug and CLion would stop at a breakpoint, but with 67b1e236334f38ec4e4d2251dbdfb790f20ed88b (adding `debug-prefix-map`) it won't stop at breakpoints anymore.
<img width="500" src="https://github.com/user-attachments/assets/66651d62-2dc9-4fa7-9f8f-d45b240aa048">

But as stated in the docs, setting the `target.source-map` in `~/.lldbini` fixed debugging again.